cmd/clusterctl/client/init_test.go

```diff
@@ -88,9 +79,88 @@ func Test_clusterctlClient_InitImages(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "returns error when core provider name is invalid",
+			args: args{
+				coreProvider:      "some-core-provider",
+				kubeconfigContext: "mgmt-context",
+			},
+			additionalProviders: []Provider{
+				config.NewProvider("some-core-provider", "some-core-url", clusterctlv1.CoreProviderType),
+			},
+			wantErr:              true,
+			expectedErrorMessage: "name cluster-api must be used with the CoreProvider type",
+		},
+		{
+			name: "return no error when core provider as the correct name",
+			args: args{
+				coreProvider:           config.ClusterAPIProviderName,
+				bootstrapProvider:      nil,
+				controlPlaneProvider:   nil,
+				infrastructureProvider: nil,
+				kubeconfigContext:      "mgmt-context",
+			},
+			expectedImages: []string{},
+			wantErr:        false,
+		},
+		{
+			name: "returns error when a bootstrap provider is not present",
+			args: args{
+				bootstrapProvider: []string{"not-provided"},
+				kubeconfigContext: "mgmt-context",
+			},
+			wantErr:              true,
+			expectedErrorMessage: "failed to get configuration for the BootstrapProvider with name not-provided",
+		},
+		{
+			name: "returns error when a control plane provider is not present",
+			args: args{
+				controlPlaneProvider: []string{"not-provided"},
+				kubeconfigContext:    "mgmt-context",
+			},
+			wantErr:              true,
+			expectedErrorMessage: "failed to get configuration for the ControlPlaneProvider with name not-provided",
+		},
+		{
+			name: "returns error when a infrastructure provider is not present",
+			args: args{
+				infrastructureProvider: []string{"not-provided"},
+				kubeconfigContext:      "mgmt-context",
+			},
+			wantErr:              true,
+			expectedErrorMessage: "failed to get configuration for the InfrastructureProvider with name not-provided",
+		},
+		{
+			name: "returns certificate manager images when required",
+			args: args{
+				kubeconfigContext: "mgmt-context",
+			},
+			wantErr: false,
+			certManagerImages: []string{
+				"some.registry.com/cert-image-1:latest",
+				"some.registry.com/cert-image-2:some-tag",
+			},
+			expectedImages: []string{
+				"some.registry.com/cert-image-1:latest",
+				"some.registry.com/cert-image-2:some-tag",
+			},
+		},
+		{
+			name: "returns error when cert-manager client cannot retrieve the image list",
+			args: args{
+				kubeconfigContext: "mgmt-context",
+			},
+			wantErr:              true,
+			certManagerImagesErr: fmt.Errorf("failed to get cert images"),
```

Comment:
minor nit: since we aren't really formatting the string.
```suggestion
			certManagerImagesErr: errors.New("failed to get cert images"),
```

Reply:
I changed that in my last push